Recruit-Training/Embedded-Recruit-Training/Core/Src/main.c

```diff
@@ -288,6 +310,27 @@ void HAL_CAN_RxFifo0MsgPendingCallback(CAN_HandleTypeDef* hcan)
     }
 
     //TODO: Match StdId of header and data length content for green and blue messages and check data for set bit of toggling green and blue led
+
+    if(hdr.DLC == 1){
+        if(BLUE_MESSAGE_STDID == hdr.StdId)
+        {
+            uint32_t bitmaskBlue = 10000101;
+            uint32_t result = data[0] + bitmaskBlue;
+            if(result == 0x85)
+            {
+                blueFlag = 1;
```

Comment:
You want to clear the flag if the bits are not set, right now you only set it to ```1```